lib/encoder.js

```diff
@@ -1442,7 +1442,7 @@ function handleBufferRef(buffer) {
  * @private
  */
 function decodeCollectionLengthV3(bytes, offset) {
-  return bytes.readUInt32BE(offset);
+  return bytes.readInt32BE(offset);
```

Comment:
I think it doesn't affect the end result in this case, but int (not uint) is the correct type according to the protocol, so it makes sense to change it: https://git-wip-us.apache.org/repos/asf?p=cassandra.git;a=blob_plain;f=doc/native_protocol_v4.spec;hb=refs/heads/trunk

I think we should also change it on `getLengthBufferV3()` when writing it.